frontends/web/src/app.tsx

```diff
@@ -191,8 +193,8 @@ class App extends Component<Props, State> {
         return prefix + ':' + JSON.stringify(this.props.devices, Object.keys(this.props.devices).sort());
     }
 
-    public componentDidUpdate(prevProps) {
-        if (prevProps.devices !== this.props.devices) {
+    public componentDidUpdate(prevProps, prevState) {
+        if (prevState.accounts !== this.state.accounts || prevProps.devices !== this.props.devices) {
```

Comment:
shouldn't hurt, but this.maybeRoute is called after the state is updated with the new account data here https://github.com/digitalbitbox/bitbox-wallet-app/blob/ab20602aca4949f0f677482adda0529a4ee5cc5d/frontends/web/src/app.tsx#L120

Reply:
Good point, I forgot about this one. reverted.